src/api.v2/events/validateAndSubmitWork.js

```diff
@@ -34,8 +35,22 @@ const validateAndSubmitWork = async (workRequest) => {
     throw new Error(`Request timed out at ${timeout}.`);
   }
 
-  const workSubmitService = new WorkSubmitService(workRequest);
+  const signedUrl = await getSignedUrl(
+    'getObject',
+    {
+      Bucket: bucketNames.WORKER_RESULTS,
+      Key: workRequest.ETag,
+    },
+  );
```

Comment:
This `signedUrl` is generated before sending to the worker so that, when the worker has the result and wants to send a signedUrl to the UI, it already has it (signed url generation from the worker was a bit problematic).

Something I would like to hopefully improve here before merging is to set a timeout for the `signedUrl` that matches that of the workRequest itself (cushioning with 60 secs more perhaps)

Reply:
If signedUrl is no longer generated from the worker, why isn't there any code connected with singed URL generation deleted in the worker PR: https://github.com/hms-dbmi-cellenics/worker/pull/335/files ? Or am I misunderstanding something?

Reply:
The `signedUrl` was never generated from the worker. 

By the phrase "signed url generation from the worker was a bit problematic" I meant that I tried to generate the signed url from the worker but it didn't work out and seemed like it wouldn't. (also it didn't feel like a good idea to generate it from many different places)